### PR TITLE
feat(gateway): Convert ILP lines to new gauge, counter, histogram schemas

### PR DESCRIPTION
**Pull request by velvia**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

In the gateway, Influx Line Protocol input was being converted to all "promCounter" schema, and histograms were being written using old Prom-style individual time series per bucket.

**New behavior :**

Influx Line Protocol inputs are now converted to gauge, counter, and histogram schemas.
summaries are not supported yet.

**BREAKING CHANGES**

Existing Prom-style histogram queries will need to be changed.

### CODE REVIEW

File: gateway/src/main/scala/filodb/gateway/conversion/InfluxProtocolParser.scala [hunk 1]
```diff
@@ -173,6 +177,15 @@ object InfluxProtocolParser extends StrictLogging {
     }
   }
 
+  // Compares the FIRST key to some bytes
+  def firstKeyEquals(bytes: Array[Byte], offsets: Buffer[Int], compareTo: Array[Byte]): Boolean = {
+    require(offsets.nonEmpty)
+    val firstKeyOffset = keyOffset(offsets(0))
+    val firstKeyLen = valOffset(offsets(0)) - 1 - firstKeyOffset
+    UnsafeUtils.equate(bytes, UnsafeUtils.arayOffset + firstKeyOffset,
```
Comment (broneill) [inline gateway/src/main/scala/filodb/gateway/conversion/InfluxProtocolParser.scala]: Can this be done without UnsafeUtils? Less unsafe is better.

Comment (velvia) [reply gateway/src/main/scala/filodb/gateway/conversion/InfluxProtocolParser.scala]: I don't think there's an equivalent here that is as performant.....

Comment (broneill) [reply gateway/src/main/scala/filodb/gateway/conversion/InfluxProtocolParser.scala]: The buffer has no feature to compare to an array?

Comment (velvia) [reply gateway/src/main/scala/filodb/gateway/conversion/InfluxProtocolParser.scala]: The only buffer is the Netty one, but by this point it's no longer visible, thus would need a whole bunch of API changes.  I'll look into it separately.

**Pull request merged**